# Graded assignment 1

Mark Klik & Misja Mikkers


# Packages

We need the package `tidyverse`

In [ ]:
#Name: Victor Digues Lacerda
#SNR: 2007801

In [1]:
library(tidyverse) 

-- Attaching packages --------------------------------------- tidyverse 1.2.1 --
<U+221A> ggplot2 2.2.1     <U+221A> purrr   0.2.4
<U+221A> tibble  1.4.2     <U+221A> dplyr   0.7.4
<U+221A> tidyr   0.8.0     <U+221A> stringr 1.3.0
<U+221A> readr   1.1.1     <U+221A> forcats 0.3.0
-- Conflicts ------------------------------------------ tidyverse_conflicts() --
x dplyr::filter() masks stats::filter()
x dplyr::lag()    masks stats::lag()


# Assignment 1

First, we generate some random data:

In [2]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1          V2        V3         V4         V5        V6        V7        
1 0.334961006 0.6391901 0.51636816 0.09354452 0.2559337 0.6724248 0.45260815
2 0.003516133 0.2630545 0.08203905 0.83706310 0.1250456 0.5269882 0.02486607
  V8        V9         V10       V11       V12        V13       V14      
1 0.1778946 0.05767805 0.1031716 0.3876952 0.08593748 0.3015531 0.1176717
2 0.1150749 0.82380932 0.9771491 0.2392356 0.69196204 0.3653892 0.5813172
  V15         V16       V17       V18        V19       V20      
1 0.528201350 0.2042594 0.5312801 0.25457674 0.1859350 0.3797968
2 0.001665078 0.1897038 0.1126481 0.07734945 0.2813688 0.4137530

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [5]:
df2 <- df1 %>%
    gather(Key, Value, V1:V20)
  

head(df2)

Key Value      
1 V1  0.334961006
2 V1  0.003516133
3 V2  0.639190058
4 V2  0.263054490
5 V3  0.516368158
6 V3  0.082039054

## Assignment 2

We will generate some data:

In [6]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [7]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df4_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [10]:
df5 <- full_join(df3, df4, by = "id")
df5

id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook